client.go

```diff
@@ -55,12 +52,28 @@ func NewClient(deployFile string, targetName TargetName, isLocalTest bool) (*Mul
 		return nil, err
 	}
 
+	appDir := path.Dir(deployFile)
+
 	// Create all SingleServerClients
 	clients := []Client{}
 	for _, target := range app.Targets(targetName) {
 		localPort := target.Base
+		var serverChannel ServerChannel
+
 		if !isLocalTest {
-			localPort = setupChannel(target.Base, target.SshPort, target.Ssh)
+			localPort = getFreeLocalPort()
+			serverChannel, err = NewSshChannel(
+				target.Base,
+				localPort,
+				target.SshPort,
+				target.Ssh,
+				newCommandRunner(appDir),
+			)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			serverChannel = NewLocalChannel(newCommandRunner(appDir))
```

Comment:
Tests run much faster (Although, this can be fixed by not killing the SSH session) - and, you don't need to have sshd running locally to run the tests (which I think is a win).